Marlin/src/inc/Conditionals_post.h

```diff
@@ -1936,206 +1928,281 @@
   #endif
 #endif
 
-//
-// Endstops and bed probe
-//
-
-#define _HAS_STOP(A,N,M,C) ((A##_HOME_TO_##M || (C+0)) && PIN_EXISTS(A##N##_##M))
+/**
+ * Endstop and probe flags
+ * - Set USE_(AXIS)_(MIN|MAX) flags for each used endstop that has a pin, including those for DIAG0 state.
+ *   - Note: Dual X Carriage uses "X" and "X2" steppers, but X_MIN and X_MAX endstop states (i.e., not X2_MAX).
+ * - Set a HAS_(AXIS)_(MIN|MAX)_STATE flag for each endstop that has a state, including SPI Sensorless which don't use a pin.
+ * - Set a HAS_(AXIS)_STATE flag for each axis that has at least one state.
+ */
+#define _USE_STOP(A,N,M,C) ((A##_HOME_TO_##M || (C+0)) && PIN_EXISTS(A##N##_##M) && !A##_SPI_SENSORLESS)
 
-#if _HAS_STOP(X,,MIN,)
-  #define HAS_X_MIN 1
+#if _USE_STOP(X,,MIN,)
+  #define USE_X_MIN 1
+#endif
+#if _USE_STOP(X,,MAX,ENABLED(DUAL_X_CARRIAGE))
+  #define USE_X_MAX 1
+#endif
+#if USE_X_MIN || BOTH(X_SPI_SENSORLESS, X_HOME_TO_MIN)
+  #define HAS_X_MIN_STATE 1
 #endif
-#if _HAS_STOP(X,,MAX,ENABLED(DUAL_X_CARRIAGE))
-  #define HAS_X_MAX 1
+#if USE_X_MAX || BOTH(X_SPI_SENSORLESS, X_HOME_TO_MAX)
+  #define HAS_X_MAX_STATE 1
 #endif
-#if HAS_X_MIN || HAS_X_MAX
-  #define HAS_X_ENDSTOP 1
+#if HAS_X_MIN_STATE || HAS_X_MAX_STATE
+  #define HAS_X_STATE 1
 #endif
 
-#if _HAS_STOP(Y,,MIN,)
-  #define HAS_Y_MIN 1
-#elif _HAS_STOP(Y,,MAX,)
-  #define HAS_Y_MAX 1
+#if _USE_STOP(Y,,MIN,)
+  #define USE_Y_MIN 1
+#elif _USE_STOP(Y,,MAX,)
+  #define USE_Y_MAX 1
 #endif
-#if HAS_Y_MIN || HAS_Y_MAX
-  #define HAS_Y_ENDSTOP 1
+#if USE_Y_MIN || BOTH(Y_SPI_SENSORLESS, Y_HOME_TO_MIN)
+  #define HAS_Y_MIN_STATE 1
+#elif USE_Y_MAX || BOTH(Y_SPI_SENSORLESS, Y_HOME_TO_MAX)
+  #define HAS_Y_MAX_STATE 1
+#endif
+#if HAS_Y_MIN_STATE || HAS_Y_MAX_STATE
+  #define HAS_Y_STATE 1
 #endif
 
-#if _HAS_STOP(Z,,MIN,EITHER(Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN, HAS_DELTA_SENSORLESS_PROBING))
-  #define HAS_Z_MIN 1
-  #if !HAS_DELTA_SENSORLESS_PROBING
-    #define HAS_Z_MIN_PIN 1
-  #endif
+#if _USE_STOP(Z,,MIN,ENABLED(Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN))
```

Comment:
Why doesn't it now depend on HAS_DELTA_SENSORLESS_PROBING?
SENSORLESS_PROBING in a delta will need USE_Z_MIN.

Reply:
The Delta sensorless state is never actually represented as an "endstop" state. See `probe_down_to_z` for the actual usage.

Reply:
Is this the part of the code you are referring to?
```
// Check to see if the probe was triggered
   const bool probe_triggered =
     #if HAS_DELTA_SENSORLESS_PROBING
       endstops.trigger_state() & (_BV(X_MAX) | _BV(Y_MAX) | _BV(Z_MAX))
     #else
       TEST(endstops.trigger_state(), Z_MIN_PROBE)
     #endif
   ;

```
I know it, and it has stopped working too. I have managed to reactivate its functionality if I use the MIN flags.